spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/UserAgentHeaderProvider.java

```diff
@@ -67,9 +62,9 @@ public String getUserAgent() {
 	private String computeUserAgent(Class clazz) {
 		String[] packageTokens = clazz.getPackage().getName().split("\\.");
 		String springLibrary = "spring-cloud-gcp-" + packageTokens[packageTokens.length - 1];
+		String version = this.getClass().getPackage().getImplementationVersion();
```

Comment:
As long as I remember you decided to stay away from this approach since someone may decide to repackage our jars without versions. See `org.springframework.core.SpringVersion` Javadocs

Reply:
We thought that many people are repackaging, but that doesn't seem to be the case. Repackaging seems to be very rare, and dealing with it is not worth the maintenance of the hard-coded versions. For example, we forgot to change the hard-coded version for 1.1 RC2 release.

Reply:
There was an issue in our data pipeline that detected the wrong version sometimes, whereas we thought it was all due to repackaging.